django/contrib/auth/backends.py

```diff
@@ -9,13 +9,17 @@ class ModelBackend(object):
     Authenticates against settings.AUTH_USER_MODEL.
     """
 
+    # Reject user when `is_active` is False.
+    allow_inactive_users = False
+
     def authenticate(self, username=None, password=None, **kwargs):
         UserModel = get_user_model()
         if username is None:
             username = kwargs.get(UserModel.USERNAME_FIELD)
         try:
             user = UserModel._default_manager.get_by_natural_key(username)
-            if user.check_password(password):
+            if (user.check_password(password) and
+                    (user.is_active or (not user.is_active and self.allow_inactive_users))):
```

Comment:
same thing here about assuming `is_active` exists and `not user.is_active` -- probably need some tests for that case.